services/navigation/builtin/builtin.go

```diff
@@ -340,6 +359,12 @@ func (svc *builtIn) AddWaypoint(ctx context.Context, point *geo.Point, extra map
 }
 
 func (svc *builtIn) RemoveWaypoint(ctx context.Context, id primitive.ObjectID, extra map[string]interface{}) error {
+	svc.mu.Lock()
+	defer svc.mu.Unlock()
+	if svc.waypointInProgress != nil && svc.waypointInProgress.ID == id && svc.navOnceCancelFunc != nil {
```

Comment:
This feels redundant; it should be impossible for `svc.navOnceCancelFunc` to be nil if `svc.waypointInProgress` is not nil, right?

Reply:
Not necessarily. In `startWaypoint`, we set the `svc.waypointInProgress` before setting `svc.navOnceCancelFunc`.